Update remote ActivityPub users when fetching their toots

ThibG opened the pull request:
On small instances, it is really frequent for remote users toots from remote non-followed users to be fetched (because of a followed user replying or boosting them), but in that case, in the ActivityPub codepath, their profile is *not* updated, which may sometime cause confusion, e.g. when a message references their avatar/name, or suggests reading their bios.

This commit changes this behavior by refreshing profile data (respecting the common 1 refresh per day per account cap) even if the user is known.

ThibG commented on app/services/activitypub/fetch_remote_status_service.rb:
```diff
@@ -44,4 +44,9 @@ def supported_context?
   end
+
+  def needs_update(actor)
+    # TODO: optimize by ruling out accounts that have local followers
```
I think the full account object may actually be present in the fetched json, but I'm not too sure it's mandated by the spec. If it is, it could be considered prefetched, avoiding further queries, and we could completely bypass the daily limit this way.

akihikodaki replied:
* It is not mandatory.
* You should _always_ fetch the account (or _dereference_ the ID in the language of the spec) to make sure the ID actually points to the account.

ThibG replied:
In that case, the account is already known, and the json is already considered authoritative for the author's toot, so I'm tempted to just use that info instead of making a new request…
But I will just trigger an account refresh for now, it can be optimized later on.

akihikodaki commented on app/services/activitypub/fetch_remote_status_service.rb:
```diff
@@ -16,7 +16,7 @@ def call(uri, id: true, prefetched_body: nil)
 
     actor = ActivityPub::TagManager.instance.uri_to_resource(actor_id, Account)
-    actor = ActivityPub::FetchRemoteAccountService.new.call(actor_id, id: true) if actor.nil?
+    actor = ActivityPub::FetchRemoteAccountService.new.call(actor_id, id: true) if actor.nil? or needs_update(actor)
```
Mastodon uses `||`

ThibG commented:
Addressed your remarks. I kind of want to avoid doing unnecessary HTTP requests if possible (and also bypass the timer in that case), but it's ok for now.
